definitions/features/smart_proxy.rb

```diff
@@ -0,0 +1,11 @@
+require 'features/foreman_proxy'
+
+class Features::SmartProxy < Features::ForemanProxy
```

Comment:
What is the difference between SmartProxy and ForemanProxy?

Reply:
Please correct me if my understanding is wrong.
I thought we have different flows to have external and in-built proxy so it thought to keep foreman_proxy feature. We can keep most of common behavior in foreman-proxy and after inheriting we can override and extend other things in smart_proxy feature (nothing but external proxy). 
I would like to know if there is scope of improvement or better way to handle possible scenarios. I have considered four:
- in-built and external smart-proxy in case of upstream
- same in case of downstream

Reply:
Ah now I see, thanks for explanation. I believe we decided to stick with Foreman Proxy everywhere to not confuse users with the two names and external proxy is called external proxy.  Do you have any specific flows on your mind? Maybe for the sake of this PR I'd keep just :foreman_proxy and :capsule and add `feature(:foreman_proxy).external?`. If we have any use case where it makes sense to split the proxy to more features we can proceed then. Does it make sense?